src/V3AstNodeOther.h

```diff
@@ -2448,6 +2451,74 @@ class AstVFile final : public AstNodeFile {
     void dumpJson(std::ostream& str = std::cout) const override;
 };
 
+// === AstNodeGen ===
+class AstGenBlock final : public AstNodeGen {
+    // Generate 'begin'
+    // @astgen op1 := genforp : Optional[AstNode]
+    // @astgen op2 := stmtsp : List[AstNode]
+    std::string m_name;  // Name of block
+    const bool m_unnamed;  // Originally unnamed (name change does not affect this)
+    const bool m_implied;  // Not inserted by user
+
+public:
+    AstGenBlock(FileLine* fl, const string& name, AstNode* stmtsp, bool implied)
+        : ASTGEN_SUPER_GenBlock(fl)
+        , m_name{name}
+        , m_unnamed{name.empty()}
+        , m_implied{implied} {
+        this->addStmtsp(stmtsp);
+    }
+    ASTGEN_MEMBERS_AstGenBlock;
+    void dump(std::ostream& str) const override;
+    void dumpJson(std::ostream& str) const override;
+    std::string name() const override VL_MT_STABLE { return m_name; }
+    void name(const std::string& name) override { m_name = name; }
+    bool unnamed() const { return m_unnamed; }
+    bool implied() const { return m_implied; }
+};
+class AstGenCase final : public AstNodeGen {
+    // Generate 'case'
+    // @astgen op1 := exprp : AstNodeExpr // Condition (scurtinee) expression
+    // @astgen op2 := itemsp : List[AstCaseItem]
```

Comment:
Should there be an GenCaseItem? Seems strange that CaseItem serves two different purposes, and can have both statements and AstGen/Procedure block items.  If you think we should go there seems better as part of this pull.

Reply:
This is a good point, especially with the aim being to strengthen types to AstNodeStmt (which we will have to do in CaseItem but can't in a GenCaseItem). I will add this.

Reply:
Thanks, forgot to say earlier feel free to merge after all of these are addressed as you feel appropriate.